pkg/security/ebpf/c/exec.h

```diff
@@ -567,6 +575,148 @@ int kprobe_exit_itimers(struct pt_regs *ctx) {
     return 0;
 }
 
+void __attribute__((always_inline)) parse_args_envs_seq(struct pt_regs *ctx, struct args_envs_seq_t *args_envs, struct str_array_ref_t *args_envs_info, u64 event_type) {
+    const char *args_start = args_envs->args_start;
+    int offset = args_envs->offset;
+    int str_count = args_envs->str_count;
+    if (str_count >= 255) {
+        return;
+    }
+    int str_total = args_envs->argc + args_envs->envc;
+
+    args_envs_info->truncated = 0;
+
+    u32 key = 0;
+    struct str_array_buffer_t *buff = bpf_map_lookup_elem(&str_array_buffers, &key);
+    if (!buff) {
+        return;
+    }
+
+    struct args_envs_event_t event = {
+        .id = args_envs_info->id,
+    };
+
+    int i = 0;
+    int n = 0;
+
+    void *perf_ptr = &buff->value[0];
+
+#pragma unroll
+    for (i = 0; i < MAX_ARRAY_ELEMENT_PER_TAIL_SEQ; i++) {
+        void *ptr = &(buff->value[(event.size + sizeof(n)) & (MAX_STR_BUFF_LEN - MAX_ARRAY_ELEMENT_SIZE - 1)]);
+
+        n = bpf_probe_read_str(ptr, MAX_ARRAY_ELEMENT_SIZE, (void *)(args_start + offset));
+        if (n > 0) {
+            n--; // remove trailing 0
+
+            // insert size before the string
+            bpf_probe_read(&(buff->value[event.size&(MAX_STR_BUFF_LEN - MAX_ARRAY_ELEMENT_SIZE - 1)]), sizeof(n), &n);
+
+            int len = n + sizeof(n);
+            if (event.size + len >= MAX_PERF_STR_BUFF_LEN) {
```

Comment:
what does perf mean here?

Reply:
Here perf refers to the eBPF perf map. MAX_PERF_STR_BUFF_LEN is the maximum length for the string buffer that we're sending in the args_envs events through the perf map.